core/types/rollup_cost.go

```diff
@@ -179,6 +193,60 @@ func NewL1CostFunc(config *params.ChainConfig, statedb StateGetter) L1CostFunc {
 	}
 }
 
+// NewOperatorCostFunc returns a function used for calculating operator fees, or nil if this is
+// not an op-stack chain.
+func NewOperatorCostFunc(config *params.ChainConfig, statedb StateGetter) OperatorCostFunc {
+	if config.Optimism == nil {
+		return nil
+	}
+	forBlock := ^uint64(0)
+	var cachedFunc operatorCostFunc
+
+	selectFunc := func(blockTime uint64) operatorCostFunc {
+		if !config.IsOptimismIsthmus(blockTime) {
+			return func(gas uint64) *uint256.Int {
+				return uint256.NewInt(0)
+			}
+		}
+		operatorFeeParams := statedb.GetState(L1BlockAddr, OperatorFeeParamsSlot).Bytes()
+		operatorFeeScalar, operatorFeeConstant := extractOperatorFeeParams(operatorFeeParams)
```

Comment:
I don't see why we should change the state array into a `[]bytes`. Let's change `extractOperatorFeeParams` to take a `common.Hash` and then we can do
```suggestion
		operatorFeeParams := statedb.GetState(L1BlockAddr, OperatorFeeParamsSlot)
		operatorFeeScalar, operatorFeeConstant := extractOperatorFeeParams(operatorFeeParams)
```
and do the early return check with `operatorFeeParams == (common.Hash{})`

Reply:
Done